content/ADR-288-deep-link-token-authentication.md

```diff
@@ -0,0 +1,433 @@
+    end
+
+    rect rgb(230, 255, 230)
+        Note over LegitUser: ✅ Security preserved:<br/>Only victim's machine can receive the token
+    end
+```
+
+**Drawbacks**:
+- Requires deep link protocol handler registration
+- May have different behavior across platforms (desktop vs. mobile)
+- Needs fallback mechanism for environments where deep links don't work
+
+**Security Analysis**: This approach is fundamentally more secure because:
+- **OS-level token delivery**: Deep links are handled by the operating system on the machine where the browser is running
+- **Prevents session fixation attacks**: Even if an attacker tricks a victim into authenticating via a malicious URL, the token is delivered to the victim's machine via OS deep link, NOT to the attacker's client
+- **Physical machine binding**: The token can only be received on the same physical machine where the browser session is active
+- **No reliance on verification codes**: The token can be a cryptographically secure random string (e.g., 256-bit) instead of a 2-digit code that can be observed or guessed
+- **Breaks the attack chain**: In the verification code flow, the attacker's client receives the AUTH CHAIN. With deep links, the attacker's client never receives the token, so it cannot complete authentication
+
+### Option 3: IP-Based Verification (Additional Security Measure)
+
+**Description**: Add server-side validation that the web session and client connection originate from the same IP address as an extra security layer.
+
+**Benefits**:
+- Simple to implement
+- No client-side changes required
+- Provides additional protection against remote attacks
+- Works transparently alongside other authentication methods
+
+**Drawbacks**:
+- Ineffective for users behind NAT, VPNs, or proxies
+- Can give false negatives (legitimate users rejected)
+- Can give false positives (multiple users behind same IP)
+- Doesn't protect against local attacks (attacker on same network)
+- Not a complete solution on its own
+
+**Status**: This is an optional additional security measure that MAY be implemented alongside Option 2 to provide defense-in-depth, but should NOT be relied upon as the primary security mechanism.
+
+### Recommended Solution
+
+**Implement Option 2 (Deep Link Token Verification) as an opt-in alternative authentication method** (via `use_token=true` parameter), while **keeping Option 1 (Verification Code) as the default authentication method**. **Option 3 (IP-Based Verification) MAY be added as an additional security layer** for defense-in-depth.
+
+## Specification
+
+### Deep Link Token Verification Flow
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
```

Comment:
This section is defined in `## RFC 2119 and RFC 8174` Would you mind removing it?